packages/lib/src/helpers/encodeCall.ts

```diff
@@ -1,15 +1,8 @@
-import abi from 'ethereumjs-abi';
-import BN from 'bignumber.js';
+import { defaultAbiCoder, ParamType } from 'ethers/utils/abi-coder';
+import { sha3 } from 'web3-utils';
 
-function formatValue(value: any): string {
-  if (typeof(value) === 'number' || BN.isBigNumber(value)) return value.toString();
-  else if (typeof(value) === 'string' && value.match(/\d+(\.\d+)?e(\+)?\d+/)) return (new BN(value)).toString(10);
-  else return value;
-}
-
-export default function encodeCall(name: string, args: string[] = [], rawValues: any[] = []): string {
-  const values: string[] = rawValues.map(formatValue);
-  const methodId: string = abi.methodID(name, args).toString('hex');
-  const params: Buffer = abi.rawEncode(args, values).toString('hex');
-  return '0x' + methodId + params;
+export default function encodeCall(name: string, types: Array<string | ParamType> = [], rawValues: any[] = []): string {
+  const encodedParameters = defaultAbiCoder.encode(types, rawValues).substring(2);
+  const signatureHash = sha3(`${name}(${types.join(',')})`).substring(2, 10);
```

Comment:
Can we add a TODO here to switch to something native of web3 once we migrate to 1.0, such as https://github.com/ethereum/web3.js/blob/1.0/packages/web3-eth-abi/src/index.js#L52?

Reply:
Moreover, when we migrate to web3 1.0, can we directly use the `web3-eth-abi`, and remove the explicit dependency on ethers.js?

Reply:
Sure! TODO note added.
Web3 1.x is actually using exactly the same dependencies we are: https://github.com/ethereum/web3.js/blob/1.0/packages/web3-eth-abi/src/index.js